### PR TITLE
Misc updates 20240703

### PR DESCRIPTION
**Pull request by dmichaels-harvard**
* Mostly changes related to additional fixes/enhancements from this doc:
   https://docs.google.com/document/d/1zj-edWR1ugqhd6ZxC07Rkq6M7I_jqiR-pO598gFg0p8
* Change to dcicutils (8.13.3.1b11) structured_data.py to NOT silently convert a
   string representing a floating point number to an integer. Per bug report here:
* Moved utility scripts view-portal-object and update-portal-object to dcicutils 8.13.3.
* Extensible validators mechanism (see submitr/validators.py) initially for submitted_id.
  See submitr/validators directory.

### CODE REVIEW

File: submitr/scripts/view_portal_object.py
Comment (dmichaels-harvard) [inline submitr/scripts/view_portal_object.py:1]: Moved to dcicutils/scripts.

File: submitr/scripts/update_portal_object.py
Comment (dmichaels-harvard) [inline submitr/scripts/update_portal_object.py:1]: Moved to dcicutils/scripts.

File: submitr/submission.py [hunk 1]
```diff
@@ -1891,11 +1910,14 @@ def progress_report(status: dict) -> None:  # noqa
                                         # i.e. this is really remove empty trailing object from arrays.
                                         remove_empty_objects_from_lists=True,
                                         # If the --merge option is given then merge the
-                                        # given (i.e. e.g. spreasheet) object(s) into any existing ones.
+                                        # given (i.e. e.g. spreadsheet) object(s) into any existing ones.
                                         merge=merge,
                                         progress=None if noprogress else define_progress_callback(debug=debug),
+                                        # Doing submitted_id validation in validate_submitted_ids, in parallel.
+                                        validator_hook=validator_hook,
```
Comment (dmichaels-harvard) [inline submitr/submission.py:1917]: This is the main generic hook into the submission metadata parsing (which currently used just to validate submission_id across all schema/types/sheets).

**Review by willronchetti — APPROVED** [review]
Looks good but you may want to test the validator hook/submitted_id validation itself in some way

**Pull request merged**